doc/port-add-del-dynamically/dynamic_port_add_del_hld.md

```diff
@@ -0,0 +1,284 @@
+8.  Portsyncd is adding a new port entry on state db.
+9.  Events from ASIC DB received on portsorch when operstate are changing (up or down).
+10. Portsorch are updating the operstate on App DB
+
+
+
+#### Del port:
+
+Del Port – Remove port element from config DB
+Note: before removing a port, the port needs to be without any dependencies (ACL, VLAN, LAG, buffer pg).
+For example: we need to remove the buffer pg that configured to a port and then remove the port.
+
+ ![Remove port](images/remove_port_diagram.png)
+
+1.  Before we remove a port, we need to remove all dependencies of these ports (vlan, acl, buffer…)
+2.  A process or a user can remove port entry from the port table on Config DB. For example, line card manager will remove port entry.
+3.  On portmgrd we receive delete event from Config DB.
+4.  Portmgrd will remove this entry on App DB.
+5.  Portsorch will receive remove entry event from the App DB.
+6.  Portsorch will delete the port and the host interface on SAI.
+7.  SAI will remove this port on SDK
+8.  Host interface will be removed and netlink event will be received on portsyncd.
+9.  Portsyncd will remove the port entry from state db
+
+
+## Modules that “listen” to changes on config port table & App port table 
+
+#### SWSS - Portsyncd:
+•   ADD PORT - Receive new port from port config table, add the port info to APP DB (update speed, interface_type, autoneg, adv_speeds, adv_interface_types).<br />
+when host interface was created add this port entry to state db<br />
+•   DEL PORT – portmgrd is removing this entry from app db.<br />
+when host interface was removed remove this port entry from state db
+
+
+#### SWSS - Portsorch:
+•   ADD PORT - Receive new port from port APP table -> create port on SAI -> create host interface -> add Flex counters<br />
+Receive notification from ASIC DB when oper_state is changing, update the port oper_state on APP db.
+•   DEL PORT - Receive del port from port APP table -> remove flex counters -> del port on SAI -> del host interface<br />
+
+Currently the orchagent is adding/removing these flex counters:
+-   PORT_BUFFER_DROP_STAT_FLEX_COUNTER_GROUP<br />
+-   PORT_STAT_COUNTER_FLEX_COUNTER_GROUP<br />
+
+
+
+Changes needs to be added:<br />
+We need to add more port counters that will be add/removed dynamically whenever port is created or removed:
+-   Queue port counters (queue & queue watermark counters)
+-   PG counters
+-   Debug counters: port ingress drops  (DEBUG_COUNTER config table)
+-   Debug counters: port egress drops 
+-   Pfc watchdog counters
+
+In the current implementation these counters were created for all ports only after init stage is done.
+
+ 
+
+
+#### PortMgrd:
+- ADD Port: Set (admin_status, mtu, learn_mode, tpid) from config db to App db 
+- Del port: Receive del port operation from port config table, remove this port from APP DB.
+
+#### Sflowmgr:
+Add port: Event from config db - Update the speed to sflow internal db.
```

Comment:
its a description of what sflowmgr is doing as a result of a change on config db ports table - it's not a change that we are planning to add.
I will try to change it in a way that it will be more clear.